lib/puppet/provider/azure_vm/azure_arm.rb

```diff
@@ -113,6 +130,34 @@ def create # rubocop:disable Metrics/AbcSize
       ip_configuration_name: default_to_name(resource[:ip_configuration_name]),
       network_interface_name: default_based_on_name(resource[:network_interface_name]),
     })
+
+    self.extensions = resource[:extensions]
+
+  end
+
+  def extensions=(value)
+    value.each do |name, properties|
+      Puppet.debug("Updating extension #{name} on vm #{resource[:name]}")
+
+      if properties.is_a?(Hash)
+        create_extension({
+          :resource_group => resource[:resource_group],
+          :location       => resource[:location],
+          :vm_name        => resource[:name],
+          :name           => name,
+          :properties     => properties,
+        })
+      elsif properties == "absent"
```

Comment:
This implementation does not match the docs, which say "Removing the extension parameter will delete the extension from the VM." One of them needs fixing, cc @kenazk